Streaming / Intro (#26)
smartdevicelink/sdl_app_library_guides: docs/Video Streaming for Navigation Apps/Introduction/index.md

# Introduction

> 

Reviewer: Should be right on the line underneath the header

> @![iOS,android]

Reviewer: JavaSE / JavaEE isn't rendering correctly (some extra text)

Mobile Navigation allows map partners to bring their applications into the car and display their maps and turn by turn easily for the user. This feature has a different behavior on the head unit than normal applications. The main differences are:

* Navigation Apps don't use base screen templates. Their main view is the video stream sent from the device.
* Navigation Apps can send audio via a binary stream. This will attenuate the current audio source and should be used for navigation commands.
* Navigation Apps can receive touch events from the video stream.
!@

@![android]

> !!! Note In order to use SDL's Mobile Navigation feature, the app must have a minimum requirement of Android 4.4 (SDK 19). This is due to using Android's provided video encoder. !!!

Reviewer: This note isn't rendering correctly

!@

@![iOS,android]
## Connecting an app
The basic connection is similar for all apps. Please follow the [Integration Basics](Getting Started/Integration Basics) guide for more information.
!@

The first difference for a navigation app is the `appHMIType` of @![iOS]`SDLAppHMITypeNavigation`!@ @![android,javaSE,javaEE]`NAVIGATION` that has to be set in the @![iOS]`SDLManger`s `SDLLifecycleConfiguration`!@ @![android,javaSE,javaEE] `SdlManagers`s `Builder`!@. Navigation apps are also non-media apps.

@![iOS]
The second difference is that a `SDLStreamingMediaConfiguration` must be created and passed to the `SDLConfiguration`. A property called `securityManagers` must be set if connecting to a version of Core that requires secure video & audio streaming. This property requires an array of classes of Security Managers, which will conform to the `SDLSecurityType` protocol. These security libraries are provided by the OEMs themselves, and will only work for that OEM. There is not a general catch-all security library.

> @!

Reviewer: This should be a `!@`

@![android]

> The second difference is the requirement to call the `setSdlSecurity(List<Class<? extends SdlSecurityBase>> secList)` method from the `SdlManager.Builder ` if connecting to an implementation of Core that requires secure video & audio streaming. This method requires an array of Security Managers, which will extend the `SdlSecurityBase` class. These security libraries are provided by the OEMs themselves, and will only work for that OEM. There is not a general catch-all security library.

Reviewer: None of this is rendering for android

* Navigation Apps can send audio via a binary stream. This will attenuate the current audio source and should be used for navigation commands.
* Navigation Apps can receive touch events from the video stream.

@![android]
!!! NOTE

> Note In order to use SDL's Mobile Navigation feature, the app must have a minimum requirement of Android 4.4 (SDK 19). This is due to using Android's provided video encoder. 

Reviewer: ```suggestion
In order to use SDL's Mobile Navigation feature, the app must have a minimum requirement of Android 4.4 (SDK 19). This is due to using Android's provided video encoder. 
```

Author: fixed

!!!
!@

## Connecting an app
The basic connection is similar for all apps. Please follow the [Integration Basics](Getting Started/Integration Basics) guide for more information.

> The first difference for a navigation app is the `appHMIType` of @![iOS]`SDLAppHMITypeNavigation`!@ @![android,javaSE,javaEE]`NAVIGATION`!@ that has to be set in the @![iOS]`SDLManager`s `SDLLifecycleConfiguration`!@ @![android,javaSE,javaEE]`SdlManagers`s `Builder`!@. Navigation apps are also non-media apps.

Reviewer: ```suggestion
The first difference for a navigation app is the `appHMIType` of @![iOS]`SDLAppHMITypeNavigation`!@ @![android,javaSE,javaEE]`NAVIGATION`!@ that has to be set in the @![iOS]`SDLManager`'s `SDLLifecycleConfiguration`!@ @![android,javaSE,javaEE]`SdlManager`'s `Builder`!@. Navigation apps are also non-media apps.
```

> The first difference for a navigation app is the `appHMIType` of @![iOS]`SDLAppHMITypeNavigation`!@ @![android,javaSE,javaEE]`NAVIGATION` that has to be set in the @![iOS]`SDLManger`s `SDLLifecycleConfiguration`!@ @![android,javaSE,javaEE] `SdlManagers`s `Builder`!@. Navigation apps are also non-media apps.

Reviewer: ```suggestion
The first difference for a navigation app is the `appHMIType` of @![iOS]`SDLAppHMITypeNavigation`!@ @![android,javaSE,javaEE]`NAVIGATION`!@ that has to be set in the @![iOS]`SDLManager`s `SDLLifecycleConfiguration`!@ @![android,javaSE,javaEE]`SdlManagers`s `Builder`!@. Navigation apps are also non-media apps.
```

Author: fixed

```
!@

@![android]
```java
SdlManager.Builder builder = new SdlManager.Builder(this, APP_ID, APP_NAME, listener);

Vector<AppHMIType> hmiTypes = new Vector<AppHMIType>();
hmiTypes.add(AppHMIType.NAVIGATION);
builder.setAppTypes(hmiTypes);

List<? extends SdlSecurityBase> securityManagers = new ArrayList();
securityManagers.add(OEMSecurityManager1.class);
securityManagers.add(OEMSecurityManager1.class);
builder.setSdlSecurity(securityManagers);

MultiplexTransportConfig mtc = new MultiplexTransportConfig(this, APP_ID, MultiplexTransportConfig.FLAG_MULTI_SECURITY_OFF);
mtc.setRequiresHighBandwidth(true);
builder.setTransportType(transport);

sdlManager = builder.build();
sdlManager.start();
```
!@

@![iOS,android]
!!! IMPORTANT
When compiling, you must make sure to include all possible OEM security managers that you wish to support.
!!!
!@

@![iOS]
## Keyboard Input
To present a keyboard (such as for searching for navigation destinations), you should use the `SDLScreenManager`'s keyboard presentation feature. For more information, see the [Popup Menus and Keyboards](Displaying a User Interface/Popup Menus and Keyboards) guide.
!@

> @![android]

Reviewer: Android and iOS should have a section for presenting menus and such using other templates.

`// TODO iOS has a section for presenting a keyboard, should android have this as well.`
!@

> @![javaEE,javaSE]

Reviewer: Move this to the top
